### ISSUE TITLE
puro crash on upgrade

### ISSUE THREAD

**Issue by RandalSchwartz** [PixelToast/puro#3]
```
Locohost.local:~ % puro upgrade master master

[E] git: error: Your local changes to the following files would be overwritten by checkout:
        bin/dart
        bin/flutter
    Please commit your changes or stash them before you switch branches.
    Aborting
[x] Bad state: git subprocess failed with exit code 1
    #0      GitClient._ensureSuccess (package:puro/src/git.dart:109)
    #1      GitClient.checkout (package:puro/src/git.dart:213)
    <asynchronous suspension>
    #2      cloneFlutterWithSharedRefs.<anonymous closure> (package:puro/src/env/create.dart:400)
    <asynchronous suspension>
    #3      ProgressNode.wrap (package:puro/src/progress.dart:62)
    <asynchronous suspension>
    #4      cloneFlutterWithSharedRefs (package:puro/src/env/create.dart:326)
    <asynchronous suspension>
    #5      upgradeEnvironment (package:puro/src/env/upgrade.dart:132)
    <asynchronous suspension>
    #6      CommandRunner.runCommand (package:args/command_runner.dart:209)
    <asynchronous suspension>
    #7      main (package:puro/src/cli.dart:224)
    <asynchronous suspension>
    
[x] Puro crashed! Please file an issue at https://github.com/PixelToast/puro
```

workaround: `puro rm master; puro create master master`

**Comment by PixelToast** [comment]
I think I know what the issue is, this only affects detached or master checkouts

You can pass `--force` in the meantime instead of deleting the affected environments

**Comment by PixelToast** [comment]
Fixed in 1.1.12
